hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/util/subject/GetSubjectNg.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.util.subject;
+
+import javax.security.auth.Subject;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+/**
+ * Indirectly calls Subject.current(), which exists in Java 18 and above only
+ */
+class GetSubjectNg implements HiddenGetSubject {
+    private final Method currentMethod;
+
+    GetSubjectNg() {
+        try {
+            currentMethod = Subject.class.getMethod("current");
```

Comment:
Use DynMethods

Reply:
This class, and all classes added in this PR, are in the hadoop-auth module because Subject.getSubject is used in both hadoop-auth (KerberosAuthenticator) and hadoop-common (UserGroupInformation).  hadoop-common depends on hadoop-auth, so that seemed like a reasonable fit.

DynMethods, however, resides in hadoop-common, and thus it cannot be used in hadoop-auth.

How would you like me to proceed?